src/table/sticky-scrollbar/sticky-scrollbar.tsx

```diff
@@ -35,17 +36,24 @@ function StickyScrollbar(
   const offsetScrollbar = hasStickyColumns || browserScrollbarSize().height === 0;
 
   useStickyScrollbar(scrollbarRef, scrollbarContentRef, tableRef, wrapperRef, offsetScrollbar);
+
+  const { ref: stickyDetectionRef, isIntersecting: isStickyDetectionVisible } = useIntersectionObserver();
```

Comment:
Is this check equivalent to the one being done [here](https://github.com/cloudscape-design/components/blob/main/src/table/sticky-scrollbar/use-sticky-scrollbar.ts#L31)?

If so, can we reuse the existing `sticky-scrollbar-visible` class instead of introducing `data-stuck`?

Or, do we need any class or attribute at all? Does this clipping style need to be conditional?

Reply:
No, that check you linked is responsible for determining whether there should be a scrollbar at all. The `sticky-scrollbar-visible` class is added if and only if the table content is wider than the available space.

The clipping style needs to be conditional, because the clipped scrollbar would otherwise show the content underneath when it is stuck:

![image](https://github.com/user-attachments/assets/4338868d-5c2b-4573-acb6-dbe35653900a)